### PR TITLE
ROC-3636: Update to Spring Boot 2 / Spring Framework 5

### PR DESCRIPTION
**Pull request by rlewan**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-3536

### Change description ###

Updating to Spring Boot 2 / Spring Framework 5.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -125,22 +128,21 @@ bintray {
...
 ext {
     reformLogging = '2.0.2'
-    springCloudVersion = 'Edgware.RELEASE'
+    springCloudVersion = 'Finchley.M9'
 }
 
 dependencies {
     compile group: 'org.springframework.boot', name: 'spring-boot-starter-actuator'
-    compile group: 'org.springframework.cloud', name: 'spring-cloud-starter-feign'
+    compile group: 'org.springframework.cloud', name: 'spring-cloud-starter-openfeign'
```
Comment (nitinprabhu) [inline build.gradle]: As you are updating the Feign client I would like to see a test where it works with `service-auth-provider-app`.i.e  you should be able to retrieve JWT token using `/lease` endpoint.

Comment (rlewan) [reply build.gradle]: What a pity there isn't one already 😉

Comment (rlewan) [reply build.gradle]: What exactly would you like to see @nitinprabhu ? Spring integration test against a Wiremock or a functional test in `service-auth-provider` service pipeline?

Comment (rlewan) [reply build.gradle]: Checking the functional/smoke tests suite in [service-auth-provider](https://github.com/hmcts/service-auth-provider-app/blob/master/api/src) and I wouldn't call it extensive 😉

Comment (nitinprabhu) [reply build.gradle]: Yes there is nothing when I was updating it if you see I released many version of `service-auth-provider-java-client` and then realised it is not working.Any test(may be even local) which uses this updated client version to access `service-auth-provider-app` and able to retrive token will do.Hope you don't have to release version of client just to test :)

Comment (timja-kainos) [reply build.gradle]: tested locally

**Pull request merged**